dockstore/dockstore-support
# Add upload-topics command to topicgenerator

**Description**
Related PR: https://github.com/dockstore/dockstore/pull/5847

This PR adds an `upload-topics` command to the topic generator so it can upload the topics generated by the `generate-topics` command to Dockstore.

**Review Instructions**
Run the `upload-topics` command against QA to upload the generated topics in this file https://github.com/dockstore/dockstore-support/blob/develop/topicgenerator/results/generated-topics_GPT_3_5_TURBO_16K_20231113T135602Z.csv
- See the topicgenerator README for instructions on how to get set up. Note that you do not need the `openai-api-key` to run the `upload-topics` command

- View this tool (should have permission to view it if you're a part of the dockstore GitHub organization) https://qa.dockstore.org/my-tools/github.com/dockstore/dockstore-tool-bamstats/bamstats_sort_cwl and verify that the AI generated topic shows up.
- Select AI Generated as the Topic Selection.
- View the tool's public page and verify that the AI topic shows up. There should be an `AI` bubble.
- Find the tool in the search table and verify that the AI topic shows up and that there's an `AI` bubble.

**Issue**
[SEAB-6007](https://ucsc-cgl.atlassian.net/browse/SEAB-6007)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install` in the project that you have modified (until https://ucsc-cgl.atlassian.net/browse/SEAB-5300 adds multi-module support properly)
- [x] If you are changing dependencies, check with dependabot to ensure you are not introducing new high/critical vulnerabilities


[SEAB-6007]: https://ucsc-cgl.atlassian.net/browse/SEAB-6007?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
